Formula/terraform.rb

```diff
@@ -35,6 +39,18 @@ def install
     system "go", "build", *std_go_args(ldflags: "-s -w")
   end
 
+  def caveats
+    <<~EOS
+      Terraform has been deprecated in homebrew-core. We will not accept any new releases.
```

Comment:
```suggestion
     We will not accept any new Terraform releases in homebrew/core.
```

as it'll already tell you it's deprecated.